src/Test/L0/SecretMaskerTests/LoggedSecretMaskerL0.cs

```diff
@@ -2,31 +2,202 @@
+
+            Assert.Equal(expectedEvents, telemetry.Count);
+
+            Dictionary<string, string> mergedDetectionData = new Dictionary<string, string>();
+
+            for (int i = 0; i < expectedDetectionEvents; i++)
+            {
+                var detectionTelemetry = telemetry[i];
+                var detectionData = detectionTelemetry.Data;
+
+                Assert.Equal(detectionTelemetry.Feature, "SecretMaskerDetections");
+
+                if (maxEventsExceeded || remainder == 0 || i < expectedDetectionEvents - 1)
+                {
+                    Assert.Equal(OssSecretMasker.MaxDetectionsPerTelemetryEvent, detectionData.Count);
+                }
+                else
+                {
+                    Assert.Equal(remainder, detectionData.Count);
+                }
+
+                foreach (var (key, value) in detectionData)
+                {
+                    Assert.True(correlatingIdsToObserve.Remove(key));
+                    Assert.Equal("TEST001/001.TestPattern", value);
+                }
+            }
+
+            if (maxEventsExceeded)
+            {
+                Assert.Equal(uniqueCorrelatingIds - OssSecretMasker.MaxTelemetryDetections, correlatingIdsToObserve.Count);
+            }
+            else
+            {
+                Assert.Equal(0, correlatingIdsToObserve.Count);
+            }
+
+            var overallTelemetry = telemetry[telemetry.Count - 1];
+            var overallData = overallTelemetry.Data;
+            Assert.Equal(overallTelemetry.Feature, "SecretMasker");
+            Assert.Equal(SecretMasker.Version.ToString(), overallData["Version"]);
+            Assert.Equal(charsScanned.ToString(CultureInfo.InvariantCulture), overallData["CharsScanned"]);
+            Assert.Equal(stringsScanned.ToString(CultureInfo.InvariantCulture), overallData["StringsScanned"]);
+            Assert.Equal(uniqueCorrelatingIds.ToString(CultureInfo.InvariantCulture), overallData["UniqueCorrelatingIds"]);
+            Assert.True(0.0 <= double.Parse(overallData["ElapsedMaskingTimeInMilliseconds"], CultureInfo.InvariantCulture));
+
+            if (maxEventsExceeded)
+            {
+                Assert.Equal("true", overallData["DetectionDataIsIncomplete"]);
+            }
+        }
     }
 
     public class VsoLoggedSecretMaskerL0 : LoggedSecretMaskerL0
     {
         protected override ISecretMasker CreateSecretMasker()
         {
-            return new SecretMasker();
+            return new VsoSecretMasker();
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "SecretMasker")]
+        public void VsoLoggedSecretMasker_TelemetryEnabled_Ignored()
+        {
+            using var lsm = new LoggedSecretMasker(CreateSecretMasker());
+            lsm.TelemetryEnabled = true;
+            Assert.False(lsm.TelemetryEnabled, "Setting TelemetryEnabled to true should be ignored since since VSO masker does not support telemetry.");
```

Comment:
```suggestion
            Assert.False(lsm.TelemetryEnabled, "Setting TelemetryEnabled to true should be ignored since VSO masker does not support telemetry.");
```